### DISCUSSION TITLE
为什么编译后,所有的i++都会变成++i?

### DISCUSSION THREAD

**Discussion by Arc-huangjingtong** [BYJRK/DotNet-Discussions#324] [C# 技术交流]
RT

**Comment by ax020913** [comment]
编译器的优化吧（i++比较++i多创建一个变量，虽然cpu只多一行的代码，cpu足够快没影响，但是大工程项目还是有一定的影响的）

就像 inline 也得看编译器的脸色来

**Comment by ax020913** [comment]
> RT
没怎么懂 RT 是什么意思，哈哈哈

**Reply by BYJRK** [reply]
“如题”

**Reply by ax020913** [reply]
哦哦，哈哈哈，了解了

**Reply by Arc-huangjingtong** [reply]
哈哈,老贴吧人了🎃

**Comment by WHYBBE** [comment]
你应该给一个例子的，让大家知道你是如何知道这个变化的

**Comment by hjslovehr** [comment]
因为 i++ 对应的底层代码是 temp = i; i+=1; return temp; 而 ++i 对应的就是 i += 1; return i; 你要是写过 cpp 的  ++  运算符重载，就能明白这个了。
https://cppreference.cn/w/cpp/language/operators
![image](https://github.com/user-attachments/assets/488b52df-ad48-490e-8f40-8535b817d8b7)

**Reply by Arc-huangjingtong** [reply]
多谢解答! 👍 🙇
